pkg/resourcegroup/controllers/resourcegroup/resourcegroup_controller.go

```diff
@@ -342,6 +352,73 @@ func (r *reconciler) computeStatus(
 	return statuses
 }
 
+// UpdateReconcileStatusToReflectKstatus uses the current Strategy and Actuation
+// status from the applier and the newly computed kstatus to compute the
+// Reconcile status. Returns an error if one of the inputs is invalid.
+func UpdateReconcileStatusToReflectKstatus(status v1alpha1.ResourceStatus) (v1alpha1.Reconcile, error) {
+	switch status.Strategy {
+	case v1alpha1.Apply:
+		switch status.Actuation {
```

Comment:
Suggest breaking these nested switches into helper functions for readability

Reply:
I left them all together to ensure it was catching all enums and permutations, but yeah, maybe adding function names would help reduce inline comments.

Reply:
I find that switches get kind of hard to follow when they are 2-3+ layers deep

Reply:
Extracted funcs for the 3rd layer. I don't think it needs it for the 2nd, but let me know if you feel differently.

Reply:
Looks good, thank you